src/watershed.jl

```diff
@@ -16,25 +16,26 @@ function _reconst_watershed(morph_residue::Matrix{<:Integer}, se::Matrix{Bool}=s
 end
 
 function watershed2(morph_residue, segment_mask, ice_mask)
+    # TODO: reconfigure to use async tasks or threads
     # Task 1: Reconstruct morph_residue
-    task1 = Threads.@spawn begin
-        mr_reconst = _reconst_watershed(morph_residue)
-        mr_reconst = ImageMorphology.local_maxima(mr_reconst; connectivity=2) .> 0
-    end
+    # task1 = Threads.@spawn begin
+    mr_reconst = _reconst_watershed(morph_residue)
+    mr_reconst = ImageMorphology.local_maxima(mr_reconst; connectivity=2) .> 0
+    # end
 
     # Task 2: Calculate gradient magnitude
```

Comment:
Maybe make an issue in the repo for threading watershed and tiling. Maybe after a hiatus, we can think about overall strategy for threading this two big operations?

Reply:
As in the threading case above, sometimes it would crash, but I'll add an issue noting where parallelism could be leveraged.

Reply:
See #523.